core/conn/jdbcT4/src/main/java/org/trafodion/jdbc/t4/InterfaceResultSet.java

```diff
@@ -503,6 +515,11 @@ static Object getExecute2FetchString(TrafT4Connection conn, TrafT4Desc desc, byt
 				break;
 			}
 			break;
+		case SQLTYPECODE_TINYINT:
+		case SQLTYPECODE_TINYINT_UNSIGNED:
+                        short sValue1 = (byte) values[noNullValue];
+			retObj = new Short(sValue1);
+			break;
```

Comment:
We dont have full support for tinyint IO from odbc/jdbc interface.
This code is part of jdbc and will not be reached until that support is in.
Once that support is in, these paths will be fixed and tested.
Until then, tinyint will be converted to smallint in the engine.

Based on the review comment, have updated this code to handle unsigned tinyint.
But it will not be tested and enabled until full jdbc/odbc support has been added.